CMakeLists.txt

```diff
@@ -273,37 +263,30 @@ if(WIN32)
             message(STATUS "Using static CRT linkage /MT")
         endif()
     else()
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mstackrealign -static-libgcc -static")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mstackrealign -static-libstdc++ -static-libgcc -static")
-        set_target_properties(${CSOUNDLIB} PROPERTIES LINK_FLAGS "-static-libstdc++ -static-libgcc -static")
-        set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -static-libstdc++ -static-libgcc -static")
+        add_compile_options("-mstackrealign")
+        add_link_options("-static"
+            "$<$<COMPILE_LANGUAGE:C>:-static-libgcc>"
+            "$<$<COMPILE_LANGUAGE:CXX>:-static-libstdc++>"
+        )
```

Comment:
I'm pretty sure `-mstackrealign` is a compile option and the others are link options but not 100% sure